packages/runtime/container-runtime/src/pendingStateManager.ts

```diff
@@ -306,6 +306,27 @@ export class PendingStateManager implements IDisposable {
 		message: InboundSequencedContainerRuntimeMessage;
 		localOpMetadata: unknown;
 	}[] {
+		if (batch.length === 0) {
+			const pendingMessage = this.pendingMessages.peekFront();
+			assert(pendingMessage !== undefined, "No pending message found for this remote message");
```

Comment:
We should assert some other stuff here to make sure it's the right shape.  This is where if we set localOpMetadata to `{ emptyBatch: true }` we could assert that.